src/lib/plot/annotation.typ

```diff
@@ -0,0 +1,64 @@
+#import "util.typ"
+#import "sample.typ"
+#import "/src/draw.typ"
+#import "/src/process.typ"
+#import "/src/util.typ"
+
+/// Add an annotation to the plot
+///
+/// An annotation is a sub-canvas that uses the plots coordinates specified
+/// by its x and y axis.
+///
+/// Bounds calculation is done naively, therefore fixed size content _can_ grow
+/// out of the plot. You can adjust the padding manually to adjust for that. The
+/// feature of solving the correct bounds for fixed size elements might be added
+/// in the future.
+///
+/// - body (drawable): CeTZ Draw commands
+/// - axes (axes): X and Y axis names
+/// - resize (bool): If true, the plots axes get adjusted to contain the annotation
+/// - padding (none,number,dictionary): Annotation padding that is used for axis
+///   adjustment
+/// - background (bool): If true, the annotation is drawn behind all plots, in the background.
+///   If false, the annotation is drawn above all plots.
```

Comment:
Could these be put into a style?

Reply:
Does not really make sense imho., as all three depend on the plot/data and are in plot units (padding).